sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -0,0 +1,132 @@
+    :ivar type: Type of event related to the originating occurrence.
+    :vartype type: str
+    :ivar time: The time (in UTC) the event was generated, in RFC3339 format.
+    :vartype time: ~datetime.datetime
+    :ivar dataschema: Identifies the schema that data adheres to.
+    :vartype dataschema: str
+    :ivar datacontenttype: Content type of data value.
+    :vartype datacontenttype: str
+    :ivar subject: This describes the subject of the event in the context of the event producer
+     (identified by source).
+    :vartype subject: str
+    :ivar specversion: Optional. The version of the CloudEvent spec. Defaults to "1.0"
+    :vartype specversion: str
+    :ivar id: An identifier for the event. The combination of id and source must be
+     unique for each distinct event. If not provided, a random UUID will be generated and used.
+    :vartype id: Optional[str]
+    """
+    def __init__(self, source, type, **kwargs): # pylint: disable=redefined-builtin
+        # type: (str, str, Any) -> None
+        self.source = source
+        self.type = type
+        self.specversion = kwargs.pop("specversion", "1.0")
+        self.id = kwargs.pop("id", str(uuid.uuid4()))
+        self.time = kwargs.pop("time", datetime.now(TZ_UTC).isoformat())
+        self.datacontenttype = kwargs.pop("datacontenttype", None)
+        self.dataschema = kwargs.pop("dataschema", None)
+        self.subject = kwargs.pop("subject", None)
+        self.extensions = {}
+        _extensions = dict(kwargs.pop('extensions', {}))
+        for key in _extensions.keys():
+            if not key.islower() or not key.isalnum() or len(key) > 20:
+                raise ValueError("Extensions must be lower case and alphanumeric.")
+        self.extensions.update(_extensions)
+        self.data = kwargs.pop("data", None)
+
+    def __repr__(self):
+        return (
+            "CloudEvent(source={}, type={}, specversion={}, id={}, time={})".format(
+                self.source,
+                self.type,
+                self.specversion,
+                self.id,
+                self.time
+            )[:1024]
+        )
+
+    @classmethod
+    def from_dict(cls, event, **kwargs):
+        # type: (Dict, Any) -> CloudEvent
+        """
+        Returns the deserialized CloudEvent object when a dict is provided.
+        :param event: The dict representation of the event which needs to be deserialized.
+        :type event: dict
+        :rtype: CloudEvent
+        """
+        data = event.pop("data", None)
+        data_base64 = event.pop("data_base64", None)
+        if data and data_base64:
+            raise ValueError("Invalid input. Only one of data and data_base64 must be present.")
+        return cls(
+        id=event.pop("id", None),
+        source=event.pop("source", None),
+        type=event.pop("type", None),
+        specversion=event.pop("specversion", None),
+        data=data or b64decode(data_base64),
+        time=event.pop("time", None),
```

Comment:
if time is datetime, need deserializaiton here

Reply:
it'd be a string